File: src/dp/house_robber.py
def robbery(houses):
    rob1, rob2 = 0, 0
    for h in houses:
        temp = max(rob1 + h, rob2)
        rob1 = rob2
        rob2 = temp
    return rob2

print(robbery([1,3,2,1])) # 4

File: src/dp/house_robber.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int robHouses(vector<int> houses) {
    int rob1 = 0, rob2 = 0;
    for (int h : houses) {
        int temp = max(h + rob1, rob2);
        rob1 = rob2;
        rob2 = temp;
    }
    return rob2;
}

int main() {

}
